### PR TITLE
chore: add dependabot to manage dependencies

### PR DESCRIPTION
**Pull request by chenrui333**
👋  I have introduced dependabot into many corporate and OSS projects (I think it would be a great tool to help managing the dependencies for this great project).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: weekly
+
+  # Maintain dependencies for dockerfile
+  - package-ecosystem: "docker"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (owenrumney) [inline .github/dependabot.yml:24]:
```suggestion
      interval: weekly
```

Looks good, does this need quotes?

Comment (chenrui333) [reply .github/dependabot.yml:24]: it is yaml config, either way is fine :)

Comment (chenrui333) [reply .github/dependabot.yml:24]: in the example, you can see it is quoted
https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#scheduleday

Comment (owenrumney) [reply .github/dependabot.yml:24]: my query was more around the inconsistency. we have 2 unquoted and 1 not

Comment (chenrui333) [reply .github/dependabot.yml:24]: I got you, let me fix it. :)

Comment (chenrui333) [reply .github/dependabot.yml:24]: all good now.

**Review by owenrumney — APPROVED** [review]

**Comment by owenrumney** [comment]
all good thanks for this

**Pull request merged**